addons/targetinfo/targetinfo.lua

```diff
@@ -72,15 +82,26 @@ windower.register_event('prerender', function()
                 '\\cs(255,0,0)' .. speed:string():lpad(' ', 5)
             or
                 '\\cs(102,102,102)' .. ('+' .. speed):lpad(' ', 5)) .. '%\\cr'
+        if mob.id == player.id then
+            info.target_name = mob.name
+        elseif mobclaim and mobclaim.id > 0 then
+            info.claim_name = mobclaim and mobclaim.name or nil
+        elseif target and target.id > 0 then
+            info.target_name = target and target.name or nil
+        else
+            remove:add('claim_name')
+            remove:add('target_name')
+        end
         text_box:update(info)
         text_box:show()
-	else
-		text_box:hide()
-	end
+        for entry in remove:it() do text_box[entry] = nil end
```

Comment:
This feature should still be optional and default to off. Also please space it out properly with new lines:
```lua
        for entry in remove:it() do
            text_box[entry] = nil
        end
```